component/host.go

```diff
@@ -29,6 +29,10 @@ type Host interface {
 	// before Component.Shutdown() begins.
 	ReportFatalError(err error)
 
+	ReportStatus(report StatusReport)
+
+	RegisterStatusReporter(reporter StatusReportFunc)
```

Comment:
1. Is there going to be an "unregister" as well?
2. What happens during restarts? Are these registrations erased automatically and the components need to re-register?